File: Ywy/Go/level1/하샤드수/harshadNumber_test.go
package harshadNumber

import (
	"testing"
)

func TestSolution(t *testing.T) {
	arr := []int{10, 12, 11, 13}
	except := []bool{true, true, false, false}
	var answer bool = false
	for index, el := range arr {
		answer = Solution(el)
		if answer != except[index] {
			t.Errorf("input: %d, answer: %v, except: %v", el, answer, except[index])
		}
	}
}

File: Ywy/JAVA/src/level1/test/cases/HarshadNumberTest.java
package level1.test.cases;

import static org.junit.Assert.assertTrue;

import org.junit.Before;
import org.junit.Test;

import level1.HarshadNumber;

public class HarshadNumberTest {
	private int[] ns;
	private boolean[] answers;
	private HarshadNumber harshadNumber;
	
	@Before
	public void setUp() {
		ns = new int[] {
			10,
			12,
			11,
			13
		};
		answers = new boolean[] {
			true,
			true,
			false,
			false
		};
		harshadNumber = new HarshadNumber();
	}
	
	@Test
	public void test() {
		int index = 0;
		int range = answers.length;
		do {
			assertTrue(harshadNumber.solution(ns[index]) == answers[index]);
		} while(++index < range);
	}

}
